Buildings/Controls/OBC/ASHRAE/G36_PR1/Generic/SetPoints/ZoneStatus.mo

```diff
@@ -1,380 +1,420 @@
-    annotation (Placement(transformation(extent={{140,-240},{180,-200}}),
-        iconTransformation(extent={{100,-110},{140,-70}})));
+    annotation (Placement(transformation(extent={{160,-220},{200,-180}}),
+        iconTransformation(extent={{100,-130},{140,-90}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput yEndSetUp
+    "True when the zone could end setup mode"
+    annotation (Placement(transformation(extent={{160,-260},{200,-220}}),
+        iconTransformation(extent={{100,-150},{140,-110}})));
 
 protected
   Buildings.Controls.OBC.CDL.Continuous.Product pro
-    "Decide if the cool down time of one zone should be ignored: if window is open, then output zero, otherwise, output cool-down time from optimal cool-down block"
-    annotation (Placement(transformation(extent={{80,210},{100,230}})));
+    "Decide if the cool down time of one zone should be ignored: if window is open, 
+    then output zero, otherwise, output cool-down time from optimal cool-down block"
+    annotation (Placement(transformation(extent={{120,210},{140,230}})));
   Buildings.Controls.OBC.CDL.Continuous.Product pro1
-    "Decide if the warm-up time of one zone should be ignored: if window is open, then output zero, otherwise, output warm-up time from optimal warm-up block"
-    annotation (Placement(transformation(extent={{80,150},{100,170}})));
-  Buildings.Controls.OBC.CDL.Continuous.Add add(final k2=-1)
-    "Calculate differential between minimum zone temperature and the heating setpoint"
-    annotation (Placement(transformation(extent={{-20,60},{0,80}})));
+    "Decide if the warm-up time of one zone should be ignored: if window is open, 
+    then output zero, otherwise, output warm-up time from optimal warm-up block"
+    annotation (Placement(transformation(extent={{120,170},{140,190}})));
+  Buildings.Controls.OBC.CDL.Continuous.Add add(
+    final k2=-1)
+    "Calculate difference between zone temperature and the occupied heating setpoint"
+    annotation (Placement(transformation(extent={{-40,80},{-20,100}})));
   Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys(
-    final pre_y_start=false,
     final uLow=-0.1,
     final uHigh=0.1)
     "Hysteresis that outputs if the system should run in warm-up mode"
-    annotation (Placement(transformation(extent={{20,60},{40,80}})));
-  Buildings.Controls.OBC.CDL.Continuous.Add add1(final k1=-1)
-    "Calculate differential between maximum zone temperature and the cooling setpoint"
-    annotation (Placement(transformation(extent={{-20,10},{0,30}})));
+    annotation (Placement(transformation(extent={{0,80},{20,100}})));
+  Buildings.Controls.OBC.CDL.Continuous.Add add1(
+    final k1=-1)
+    "Calculate difference between zone temperature and the occupied cooling setpoint"
+    annotation (Placement(transformation(extent={{-40,0},{-20,20}})));
   Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys1(
-    final pre_y_start=false,
     final uLow=-0.1,
     final uHigh=0.1)
```

Comment:
- uLow and uHigh settings of hys and hys1 should be exposed as parameters for this block so that the user can change them, for example, when coupling with optimal start block. uLow of hys and hys1 can be set as one parameter, so does uHigh of hys and hys1.
- I suggest to use two parameters separately for uLow and uHigh, instead of using one parameter "bouLim" and setting uLow=-0.5*bouLim and uHigh=0.5*bouLim. The reason why I suggest this is because in the optimal start, I set the default uLow to be non-negative. uLow is used to judge when the zone temperature reaches the setpoint, and setting uLow to be zero helps to narrow down the deviation between zone temperature and setpoint (comparing with -0.5). Setting uLow to positive helps to mitigate corner cases that the zone temperature takes too long or never reaches the setpoint. Please correct me if I am wrong here.